File: CPP/PemilikRuko.cpp
#pragma once
#include <iostream>
#include <string>
#include <list>
#include "Orang.cpp"
#include "Ruko.cpp"

using namespace std;


class PemilikRuko : public Orang
{
private:
    string NPWP;
    string rekeningBank;
    int lamaMenjadi;
    vector<Ruko> listRuko;
public:
    PemilikRuko()
    {


    }


    PemilikRuko(string nama, int usia, string alamat, string noTelp, string NPWP, string rekeningBank, int lamaMenjadi, vector<Ruko> listRuko) : Orang(nama, usia, alamat, noTelp)
    {
        this->NPWP = NPWP;
        this->rekeningBank = rekeningBank;
        this->lamaMenjadi = lamaMenjadi;
        this->listRuko = listRuko;
    }

    // Setter methods
    void setNPWP(string NPWP) {
        this->NPWP = NPWP;
    }

    void setRekeningBank(string rekeningBank) {
        this->rekeningBank = rekeningBank;
    }

    void setLamaMenjadi(int lamaMenjadi) {
        this->lamaMenjadi = lamaMenjadi;
    }

    void setListRuko(vector<Ruko> listRuko) {
        this->listRuko = listRuko;
    }

    // Getter methods
    string getNPWP() {
        return this->NPWP;
    }

    string getRekeningBank() {
        return this->rekeningBank;
    }

    int getLamaMenjadi() {
        return this->lamaMenjadi;
    }

    vector<Ruko> getListRuko() {
        return this->listRuko;
    }

    void addRuko(Ruko ruko) {
        this->listRuko.push_back(ruko);
    }
    
    ~PemilikRuko()
    {


    }
};

File: PY/PemilikRuko.py
from Orang import Orang
from Ruko import Ruko

class PemilikRuko(Orang):
    __NPWP = ""
    __rekeningBank = ""
    __lamaMenjadi = 0
    __listRuko = []
    
    def __init__(self, nama="", usia=0, alamat="", noTelp="", NPWP="", rekeningBank="", lamaMenjadi=0, listRuko=[]):
        super().__init__(nama, usia, alamat, noTelp)
        self.__NPWP = NPWP
        self.__rekeningBank = rekeningBank
        self.__lamaMenjadi = lamaMenjadi
        self.__listRuko = listRuko

    # Setter dan Getter untuk NPWP
    def setNpwp(self, NPWP):
        self.__NPWP = NPWP
        
    def getNpwp(self):
        return self.__NPWP
    
    # Setter dan Getter untuk rekeningBank
    def setRekeningBank(self, rekeningBank):
        self.__rekeningBank = rekeningBank
        
    def getRekeningBank(self):
        return self.__rekeningBank
    
    # Setter dan Getter untuk lamaMenjadi
    def setLamaMenjadi(self, lamaMenjadi):
        self.__lamaMenjadi = lamaMenjadi
        
    def getLamaMenjadi(self):
        return self.__lamaMenjadi
    
    # Setter dan Getter untuk listRuko
    def setListRuko(self, listRuko):
        self.__listRuko = listRuko
        
    def getListRuko(self):
        return self.__listRuko

    def addRuko(self, ruko):
        self.__listRuko.append(ruko)
